spring-integration-r2dbc/src/main/java/org/springframework/integration/r2dbc/outbound/R2dbcMessageHandler.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.r2dbc.outbound;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.data.r2dbc.core.DatabaseClient;
+import org.springframework.data.r2dbc.core.R2dbcEntityOperations;
+import org.springframework.data.relational.core.query.Criteria;
+import org.springframework.data.relational.core.query.Update;
+import org.springframework.data.relational.core.sql.SqlIdentifier;
+import org.springframework.expression.Expression;
+import org.springframework.expression.TypeLocator;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.expression.spel.support.StandardTypeLocator;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.handler.AbstractReactiveMessageHandler;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+import reactor.core.publisher.Mono;
+
+
+/**
+ * Implementation of {@link org.springframework.messaging.ReactiveMessageHandler} which writes
+ * Message payload into a Relational Database, using reactive r2dbc support.
+ *
+ * @author Rohan Mukesh
+ *
+ * @since 5.4
+ */
+public class R2dbcMessageHandler extends AbstractReactiveMessageHandler {
+
+	private R2dbcEntityOperations r2dbcEntityOperations;
+
+	private volatile boolean initialized = false;
+
+	private StandardEvaluationContext evaluationContext;
+
+	private Expression tableNameExpression;
+
+	@Nullable
+	private Expression valuesExpression;
+
+	@Nullable
+	private Expression criteriaExpression;
+
+	private Expression queryTypeExpression = new ValueExpression<>(Type.INSERT);
+
+
+	/**
+	 * Construct this instance using a fully created and initialized instance of provided
+	 * {@link R2dbcEntityOperations}
+	 *
+	 * @param r2dbcEntityOperations The R2dbcEntityOperations implementation.
+	 */
+
+	public R2dbcMessageHandler(R2dbcEntityOperations r2dbcEntityOperations) {
+		Assert.notNull(r2dbcEntityOperations, "'r2dbcEntityOperations' must not be null");
+		this.r2dbcEntityOperations = r2dbcEntityOperations;
+	}
+
+
+	public void setQueryType(R2dbcMessageHandler.Type type) {
```

Comment:
We will need to add javadocs to all the public API eventually.
This is open source, so would be great to give end-users as much useful info as possible! 😄